### ISSUE TITLE
libc not found if path differs

### ISSUE THREAD

**Issue by Cre3per** [frida/frida-core#364]
I wasn't able to figure out what causes the path to differ, sorry for the lack of information up ahead.

```
Host           Arch Linux x86_64
Application PE32+ executable (GUI) x86-64, for MS Windows (Old School Rune Scape, osclient.exe), Running through Proton 5.13-6
Frida          14.2.13
```

The application loads various libraries from `/run/host/usr/lib/`, rather than `/usr/lib/` directly.  
This causes [`frida_find_library_base()`](https://github.com/frida/frida-core/blob/c7bb50e419af3c0e42ccc7b1eeba39f5cad45553/src/linux/frida-helper-backend-glue.c#L2976) in [`frida_resolve_library_function()`](https://github.com/frida/frida-core/blob/c7bb50e419af3c0e42ccc7b1eeba39f5cad45553/src/linux/frida-helper-backend-glue.c#L2967) to fail, because `frida_find_library_base()` is called with the host's library path (`/usr/lib/libc-2.33.so`).

Error message `Failed to attach: unable to inject library into process without libc`.

Allowing the paths to differ by changing

```
remote_base = frida_find_library_base (pid, local_library_path, &remote_library_path);
// to
remote_base = frida_find_library_base (pid, library_name, &remote_library_path);
```

and disabling the assertion below gets rid of the error and frida works as expected (Tested with `NativePointer::readU8()` and `Process.enumerateModules()` in node).

Obviously this isn't a valid fix, because the host and client library have to be identical. Unfortunately, the directory `/run/host/` doesn't exist to verify that the libraries are identical, I suppose there's some redirection happening in Proton/wine.

I don't see a way to solve this automatically without digging through Proton/wine and writing a specific solution to that.  
I suggest adding an option (Command line, bindings, environment, ...) to allow library paths to differ. Either by enabling the changes I did above, or by specifying a library path prefix, eg. `--library-prefix /run/host/`, and looking up the library there if the first call to `frida_find_library_base()` fails.

**Comment by oleavr** [comment]
Yeah it would be great to fix this. One solution that I've been thinking about for a while is to write some position-independent and highly constrained C code that we use to do the probing from the inside of the target process. This worked out really well for jailed iOS, where we have some position-independent [helpers](https://github.com/frida/frida-core/tree/master/src/fruity/helpers) that we write into memory and execute. It may sound a bit complex, but it will allow us to also support injecting into fully static programs, as we can bootstrap the dynamic linker and then load our agent. (In theory, I haven't explored this yet.)
